examples/chip-tool/commands/common/ModelCommandResponse.h

```diff
@@ -0,0 +1,41 @@
+/*
+ *   Copyright (c) 2020 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#pragma once
+
+class ModelCommandResponse
+{
+public:
+    ModelCommandResponse(uint8_t commandId) { mCommandId = commandId; }
+
+    bool HandleCommandResponse(uint8_t commandId, uint8_t * message, uint16_t messageLen)
+    {
+        VerifyOrExit(
```

Comment:
nit, feel free to ignore.

I generally find the implied 'goto' odd, especially if 2 returns exist anyway. How about:

```
if (commandId != mCommandId) {
  ChipLogError(chipTool, "Unexpected command response id: '0x%02x'. Expecting: '0x%02x'", commandId, mCommandId));
  return false;
}
return HandleResponse(message, messageLen);
```

I personally find the forcing of VerifyOrExit everywhere odd ... would be curious what others think.

</endrant> ... just wanted to get this complaint out there.

Reply:
> I generally find the implied 'goto' odd, especially if 2 returns exist anyway.

I admit that in this case, if there are multiple returns it looks odd.
> I personally find the forcing of VerifyOrExit everywhere odd ... would be curious what others think.
> 
> ... just wanted to get this complaint out there.

For my part I do like early returns. But its true that when there are many returns, for examples when you are doing a bunch of checks against parameters or multiple returns values, the CodeUtils macros makes the code more readable.

I would say that in some cases it would be better to have early returns or something else but I guess the core idea behind enforcing the CodeUtils macro is code consistency.

Reply:
(Fwiw I removed the use of VerifyOrExit here)